pkg/email/send_email.go

```diff
@@ -0,0 +1,44 @@
+package email
+
+import (
+	"github.com/treeverse/lakefs/pkg/email/params"
+	gomail "gopkg.in/gomail.v2"
+)
+
+func NewEmail(e params.Email, err error) (*params.Email, error) {
+	email := &params.Email{
+		SMTPHost:           e.SMTPHost,
+		Port:               e.Port,
+		Username:           e.Username,
+		Password:           e.Password,
+		Sender:             e.Sender,
+		Receivers:          e.Receivers,
+		Subject:            e.Subject,
+		Body:               e.Body,
+		AttachmentFilePath: e.AttachmentFilePath,
+	}
+	return email, err
+}
+
+func SendEmail(e params.Email) error {
+	msg := gomail.NewMessage()
+	msg.SetHeader("From", e.Sender)
+	msg.SetHeader("To", e.Receivers...)
+	msg.SetHeader("Subject", e.Subject)
+	msg.SetBody("text/html", e.Body)
+	if len(e.AttachmentFilePath) > 0 {
+		for _, f := range e.AttachmentFilePath {
+			msg.Attach(f)
+		}
+	}
+	d := gomail.NewDialer(e.SMTPHost, e.Port, e.Username, e.Password)
+	return d.DialAndSend(msg)
+}
```

Comment:
From what I understand, to send an email I would need to initiate a `params.Email` struct with all the fields (such as `SMTPHost`, `Port`, etc...) and then call SendEmail with the struct.

I would expect something like 
```
NewMailer(SmtpHost, Port, Username, Password, Sender) &Mailer
```
And then send mail using 
```
mailer.mail(Receivers, Subject, Body)
```
That way we could initiate lakeFS with some Sender and then sending a mail all I need is Receivers, Subject and body

So for example, to send an email from the controller I would need to run something like
```
c.mailer.mail(receivers, subject, body)
```

Reply:
Such a struct is automatically created during the init of the Lakefs config file, so the assumption is that such a struct exists and is initialized already and all that is needed is to add params such as body or subject, and for that, there is the function of `InsertAdditionalParamsToEmail` to give that capability.